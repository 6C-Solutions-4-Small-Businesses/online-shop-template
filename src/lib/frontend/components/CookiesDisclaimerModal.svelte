<script lang="ts">
    import {wereCookiesAccepted} from '$lib/frontend/stores/localStorageStore/AcceptCookiesStore'
    import {showCookiesDisclaimer} from "$lib/frontend/stores/localStorageStore/ShowCookiesDisclaimerStore"
    import {t} from '$translations/index'

    wereCookiesAccepted.subscribe(value => {
        if (!value) {
            showCookiesDisclaimer.set(true)
        }
    })

    const updateCookies = (accepted: boolean) => {
        wereCookiesAccepted.set(accepted)
        showCookiesDisclaimer.set(false)
    }

</script>

<div class="fixed bottom-0 p-1 right-0 bg-green-50 shadow-md cookies-container block}">
    <div class="border p-4 border-gray-600">
        <p class="font-light text-xl">{$t('lib/frontend/components/cookies-disclaimer.title')}</p>

        <br>

        <p class="font-thin">
            {$t('lib/frontend/components/cookies-disclaimer.body')}
        </p>

        <br>

        <p class="font-thin">
            {$t('lib/frontend/components/cookies-disclaimer.note')} <a href="./privacy-policies">{$t('lib/frontend/components/cookies-disclaimer.link')}.</a>
        </p>

        <div class="flex justify-center mt-4">
            <button data-testid="decline" class="btn variant-soft rounded-none mr-5"
                    on:click={() => updateCookies(false)}>
                {$t('lib/frontend/components/cookies-disclaimer.decline')}
            </button>

            <button data-testid="accept" class="btn variant-filled rounded-none" on:click={() => updateCookies(true)}>
                {$t('lib/frontend/components/cookies-disclaimer.accept')}
            </button>
        </div>
    </div>
</div>

<style lang="css">
    .cookies-container {
        min-height: 296px;
        max-width: 400px;
    }
</style>
