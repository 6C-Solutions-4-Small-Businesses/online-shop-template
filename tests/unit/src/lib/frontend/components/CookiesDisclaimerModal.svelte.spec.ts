import {render} from "@testing-library/svelte"
import CookiesDisclaimerModal from "$lib/frontend/components/CookiesDisclaimerModal.svelte"

describe('Cookies Disclaimer', () => {

    describe('statically', () => {

        it('should have matching snapshot', () => {

            const {container} = render(CookiesDisclaimerModal)

            expect(container).toMatchSnapshot()
        })
    })
})
