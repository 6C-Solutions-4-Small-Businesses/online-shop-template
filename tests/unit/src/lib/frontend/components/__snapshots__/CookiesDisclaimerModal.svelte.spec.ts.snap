// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cookies Disclaimer > statically > should have matching snapshot 1`] = `
<body>
  <div>
    <div
      class="fixed bottom-0 p-1 right-0 bg-green-50 shadow-md cookies-container block} s-_HEqA1ltobFh"
    >
      <div
        class="border p-4 border-gray-600 s-_HEqA1ltobFh"
      >
        <p
          class="font-light text-xl s-_HEqA1ltobFh"
        >
          
        </p>
         
        <br
          class="s-_HEqA1ltobFh"
        />
         
        <p
          class="font-thin s-_HEqA1ltobFh"
        >
          
        </p>
         
        <br
          class="s-_HEqA1ltobFh"
        />
         
        <p
          class="font-thin s-_HEqA1ltobFh"
        >
          
           
          <a
            class="s-_HEqA1ltobFh"
            href="./privacy-policies"
          >
            
            .
          </a>
        </p>
         
        <div
          class="flex justify-center mt-4 s-_HEqA1ltobFh"
        >
          <button
            class="btn variant-soft rounded-none mr-5 s-_HEqA1ltobFh"
            data-testid="decline"
          >
            
          </button>
           
          <button
            class="btn variant-filled rounded-none s-_HEqA1ltobFh"
            data-testid="accept"
          >
            
          </button>
        </div>
      </div>
    </div>
    <!--&lt;CookiesDisclaimerModal&gt;-->
  </div>
</body>
`;
